opensearch-notebooks/build.gradle

```diff
@@ -29,7 +29,10 @@ import java.util.concurrent.Callable
 
 buildscript {
     ext {
-        opensearch_version = System.getProperty("opensearch.version", "1.0.0")
+        opensearch_version = System.getProperty("opensearch.version", "1.1.0-SNAPSHOT")
+        // 1.0.0 -> 1.0.0.0, and 1.0.0-SNAPSHOT -> 1.0.0.0-SNAPSHOT
+        opensearch_build = opensearch_version.replaceAll(/(\.\d)([^\d]*)$/, '$1.0$2')
```

Comment:
Why do we want to implicitly morph the incoming version with an extra `.0` ?

Reply:
OpenSearch is a 3-digit version, so you start with 1.1.0 or 1.1.0-SNAPSHOT, but common utils and plugins are 4-digit, so 1.1.0.0 and 1.1.0.0-SNAPSHOT. We have https://github.com/opensearch-project/OpenSearch/issues/1093 open to stop doing that.

Reply:
This is just a clever(er) way of doing `opensearch_version - '-SNAPSHOT' + '.0' + ('-SNAPSHOT' if opensearch_version.contains('-SNAPSHOT') else '').